scripts/build/build_darwin_framework.py

```diff
@@ -75,6 +70,38 @@ def build_darwin_framework(args):
         command += [
             "CHIP_INET_CONFIG_ENABLE_IPV4=NO",
         ]
+
+    if not args.asan and not args.tsan:
+        # For now disable unguarded-availability-new warnings because we
+        # internally use APIs that we are annotating as only available on
+        # new enough versions.  Maybe we should change out deployment
+        # target versions instead?
+        command += [
+            "OTHER_CFLAGS=${inherited} -Wno-unguarded-availability-new",
+        ]
+
+    if args.asan:
+        command += [
+            "CHIP_IS_ASAN=YES",
+            "OTHER_LDFLAGS=${inherited} -fsanitize=address -fno-omit-frame-pointer",
+            "OTHER_CFLAGS=${inherited} -Wno-unguarded-availability-new -fsanitize=address -fno-omit-frame-pointer",
```

Comment:
As a followup: Maybe we should build up arrays of cflags/ldflags based on the conditionals, then have a single append to `command`?  Something like:

```
    cflags = ["OTHER_CFLAGS=${inherited} -Wno-unguarded-availability-new"]
    ldflags = " ["OTHER_LDFLAGS=${inherited}"]
    if args.asan:
        asan_flags = "-fsanitize=address -fno-omit-frame-pointer"
        cflags.append(asan_flags)
        ldlflags.append(asan_flags)
   elif args.tsan: # Or throw if both are set, since that's not valid, instead of waiting for the compile to fail?
        tsan_flags = ...
        # Append them here

   command += [ cflags.join(' '), ldflags.join(' ') ]
```
or something like that.  Just to reduce the repetition.